### PR TITLE
Implemented Move class.

### PR DESCRIPTION
**Pull request by datGUUYY**

### CODE REVIEW

File: idk/src/main/java/com/treydev/idk/Critter.java [hunk 1]
```diff
@@ -3,32 +3,44 @@
 import java.util.ArrayList;
 
 public class Critter {
-    private static ArrayList<Critter> critters = new ArrayList<Critter>();
+    //private static ArrayList<Critter> critters = new ArrayList<Critter>();
```
Comment (dmckinstry) [inline idk/src/main/java/com/treydev/idk/Critter.java:6]: I notice you removed the critter list.  Are you going to do away with it?  Or do you still plan to have a "dictionary" of all valid critters stored in such a list?  If you don't need it anymore, consider just deleting the line.

File: idk/src/main/java/com/treydev/idk/Critter.java [hunk 2]
```diff
@@ -3,32 +3,44 @@
...
-    public int baseDefense;
+    public int health;
+    public Move[] moveset;
 
-    private Critter(String name, int baseAttack, int baseDefense)
+    public Critter(String name, int baseAttack, int Health, Move[] moveset)
     {
         this.baseAttack = baseAttack;
-        this.baseDefense = baseDefense;
+        this.health = Health;
```
Comment (dmckinstry) [inline idk/src/main/java/com/treydev/idk/Critter.java:16]: I understand health...  Do creatures have also have a defense/armor rating?

File: idk/src/main/java/com/treydev/idk/Critter.java [hunk 3]
```diff
@@ -3,32 +3,44 @@
 import java.util.ArrayList;
```
Comment (dmckinstry) [inline idk/src/main/java/com/treydev/idk/Critter.java:3]: I think differentiating Species from Critter might be a good next step.

File: idk/src/main/java/com/treydev/idk/PhysicalAttack.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package com.treydev.idk;
+
+import java.util.ArrayList;
```
Comment (dmckinstry) [inline idk/src/main/java/com/treydev/idk/PhysicalAttack.java:3]: General cleanup - consider removing whitespace between imports

File: idk/src/main/java/com/treydev/idk/PhysicalAttack.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+import jakarta.security.auth.message.MessagePolicy.Target;
+
+public abstract class PhysicalAttack extends Move {
```
Comment (dmckinstry) [inline idk/src/main/java/com/treydev/idk/PhysicalAttack.java:7]: In general, consider adding some class header comments

File: idk/src/main/java/com/treydev/idk/HelloWorldController.java [hunk 6]
```diff
@@ -50,8 +50,10 @@ public String game(@RequestParam("name") String name, @RequestParam("action") St
         return "Game";
     }
 
+    //!!! These are the persistent memory items for temporary stubs
```
Comment (dmckinstry) [inline idk/src/main/java/com/treydev/idk/HelloWorldController.java:53]: Consider renaming HelloWorldController to something more meaningful... GameController?  I think in the long run you'll separate into different controller classes, potentially in a different subfolder

**Review by dmckinstry — APPROVED** [review]
I think you can merge it, but consider a little cleanup before doing so...

File: idk/src/main/java/com/treydev/idk/Critter.java (hunk 2 above)
Comment (datGUUYY) [reply idk/src/main/java/com/treydev/idk/Critter.java:16]: They will, and a few more stats, but that's going to be in the implementation of "Species" mostly.

File: idk/src/main/java/com/treydev/idk/Critter.java [hunk 7]
```diff
@@ -21,7 +21,7 @@ public Critter(String name, int baseAttack, int Health, Move[] moveset)
     public static ArrayList<Critter> initializeStubbedParty()
     {
         ArrayList<Critter> value = new ArrayList<>();
-        value.add(new Critter("Widget", 10, 5, Move.implementStubbedMoveList()));
+        value.add(new Critter("Widget", Species.stubWidget(), 10, 5, Move.implementStubbedMoveList()));
```
Comment (dmckinstry) [inline idk/src/main/java/com/treydev/idk/Critter.java]: I think we'll want to get rid of the specific per-species stub methods and try to make sure the actual stubbing happens hidden, inside of the Species class and not referenced from the calling class.  I'm concerned that this pattern means you'll develop separate methods for each species but I thought you wanted random species.  (Could just be my misunderstanding of your intent.)

**Pull request merged**